packages/aws-cdk-lib/aws-ec2/lib/instance.ts

```diff
@@ -398,14 +407,22 @@ export class Instance extends Resource implements IInstance {
       });
     }
 
+    // network interfaces array is set to configure the primary network interface if associatePublicIpAddress is true or false
+    const networkInterfaces = props.associatePublicIpAddress !== undefined
```

Comment:
I wonder if we should add `DeleteOnTermination`. The docs don't state what the default is so I'm not sure. What do you think?

Reply:
This is a good point. I just launched an instance via the console and the primary network interface has DeleteOnTermination set to true. To ensure we get the same behavior I think it would be good to explicitly set this as true.